jwql/utils/utils.py

```diff
@@ -301,6 +301,15 @@ def filename_parser(filename):
         r"_(?P<detector>((?!_)[\w])+)"\
         r"_(?P<ac_id>(o\d{3}|(c|a|r)\d{4}))"
 
+    # Stage 2 MSA metadata file. Created by APT and loaded in
+    # assign_wcs. e.g. "jw01118008001_01_msa.fits"
+    stage_2_msa = \
+        r"jw" \
+        r"(?P<program_id>\d{5})"\
```

Comment:
I know they occur throughout the entire code base, but it makes sense to bring it up now if its an easy enough fix.  We should avoid using "magic numbers" in our code if we don't have to.  "Constants" in python do not exist, so to let the programmer know that the "variable" should not be changed you can write it in ALL_CAPS.  So a solution these magic numbers would be to append our magic numbers to our currently existing constants.py module, and store them there.  Something like:

```
# Filepath Name Lengths
PROGRAM_ID_LEN = 5
OBSERVATION_LEN = 3
VISIT_LEN = 3
```

import this new constants.py module where its needed, and use the defines appropriately
```
r"(?P<program_id>\d{PROGRAM_ID_LEN})"\
r"(?P<observation>\d{OBSERVATION_LEN})"\
r"(?P<visit>\d{VISITS_LEN})"\
```


Once established we can use constants.py to clean up other magic numbers as we encounter them.

Reply:
Done. Perhaps at the expense of a bit of readability. But then regular expressions are not the most readable things to begin with.